Docs update (#406)
elastic/apm-agent-android: docs/reference/getting-started.md

---
mapped_pages:
  - https://www.elastic.co/guide/en/apm/agent/android/current/setup.html
---

> # Getting started

Reviewer: I was a bit lost here in the beginning as we jump right into the action. I would like propose to add an introductorily section to outline what the reader has to expect on this page like a short summary of what is coming.

Author: Make sense. I've just added an intro, let me know what you think 👍

Reviewer: Thanks

---
mapped_pages:
  - https://www.elastic.co/guide/en/apm/agent/android/current/setup.html
---

# Getting started

> By the time you finish this guide, you'll have a fully set-up agent in your app and know how to create and explore your app's data in {{kib}}!

Reviewer 2: ```suggestion
Learn how to set up an agent in your app and explore data from your app in {{kib}}!
```

## Requirements

| Requirement                                       | Minimum version                                                                                           |
|---------------------------------------------------|-----------------------------------------------------------------------------------------------------------|
| [{{stack}}](https://www.elastic.co/elastic-stack) | 8.18                                                                                                      |
| Android Gradle plugin                             | 7.4.0                                                                                                     |
| Android API level                                 | 26 (or 21 with [desugaring](https://developer.android.com/studio/write/java8-support#library-desugaring)) |

:::{important} - Apps with minSdk below 26
If your application's [minSdk](https://developer.android.com/studio/publish/versioning#minsdk) value is lower than 26, you **must** add [Java 8 desugaring support](https://developer.android.com/studio/write/java8-support#library-desugaring).

More info on [FAQs](faq.md#why-desugaring).
:::

## Gradle setup

> Add the [Elastic OTel Agent plugin](https://plugins.gradle.org/plugin/co.elastic.otel.android.agent) to your application’s `build.gradle[.kts]` file, as shown below:

Reviewer 2: ```suggestion
Add the [Elastic OTel Agent plugin](https://plugins.gradle.org/plugin/co.elastic.otel.android.agent) to your application’s `build.gradle[.kts]` file:
```

```kotlin
plugins {
    id("com.android.application")
    id("co.elastic.otel.android.agent") version "[latest_version]" // <1>
}
```

1. You can find the latest version [here](https://plugins.gradle.org/plugin/co.elastic.otel.android.agent).

## Agent setup

> Once the gradle setup is done, you need to initialize the agent within your app's code, as shown below:

Reviewer 2: ```suggestion
After Gradle setup is done, initialize the agent within your app's code:
```

```kotlin
val agent = ElasticApmAgent.builder(application) // <1>
    .setServiceName("My app name") // <2>
    .setExportUrl("http://10.0.2.2:4318") // <3>
    .setExportAuthentication(Authentication.ApiKey("my-api-key")) // <4>
    .build()
```

1. Your [Application](https://developer.android.com/reference/android/app/Application) object. Check out [how to get it](how-tos.md#get-application).

> 2. "Service" is OpenTelemetry's jargon for an "entity that produces telemetry", so here's where your application name should go. More info on [FAQs](faq.md#why-service).

Reviewer 2: ```suggestion
2. In OpenTelemetry, _service_ means _an entity that produces telemetry_, so this is where your application name should go. For more information, go to [FAQs](faq.md#why-service).
```

3. This is the Elastic endpoint where all your telemetry will be exported. If you don't have one yet, check out [how to get it](how-tos.md#get-export-endpoint).

> 4. Using an API Key is the recommended authentication method for the agent to connect to your {{stack}}. If you don't have one yet, check out [how to create one](how-tos.md#create-api-key).

Reviewer 2: ```suggestion
4. Use an API key to connect the agent to the {{stack}}. If you don't have an API key yet, check out [how to create one](how-tos.md#create-api-key).
```

:::{include} _snippets/tip-provide-values-from-outside.md
:::

## Hello World!

The agent is fully initialized, so now you can start sending telemetry to your {{stack}}! Here's a quick example of manually creating a [span](https://opentelemetry.io/docs/concepts/signals/traces/#spans) and finding it in {{kib}}:

### Generate telemetry

```kotlin
val agent = ElasticApmAgent.builder(application)
    .setServiceName("My app name")
    //...
    .build()

agent.span("My Span") {
    Thread.sleep(500) // <1>
    agent.span("My nested Span") { // <2>
        Thread.sleep(500) 
    }
}
```
1. This is to simulate some code execution for which we want to measure the time it takes to complete.

> 2. This is to demonstrate how span hierarchies look like in {{kib}}.

Reviewer 3: ```suggestion
2. This is to demonstrate what span hierarchies look like in {{kib}}.
```

### Visualize telemetry

Once your app has sent telemetry data, either [manually](manual-instrumentation.md) or [automatically](automatic-instrumentation.md), you should be able to visualize it in {{kib}} by navigating to **Applications -> Service Inventory** in the main menu, or alternatively, searching for "Service Inventory" in the [global search field](https://www.elastic.co/guide/en/kibana/current/introduction.html#kibana-navigation-search).

You should find your application listed there.

:::{image} ../images/span-visualization/1.png
:screenshot:
:width: 350px
:::

When you open it, go to the **Transactions** tab, where you should see your app's "outermost" spans listed.

:::{image} ../images/span-visualization/2.png
:screenshot:
:width: 350px
:::

> And after clicking on the span, you should see it in detail.

Reviewer 3: ```suggestion
After clicking on the span, you should see it in detail.
```

> You should find your application listed there, as shown below:
> 
> :::{image} ../images/span-visualization/1.png
> :screenshot:
> :width: 350px
> :::
> 
> When you open it, **go to the "Transactions" tab**, where you should see your app's "outermost" spans listed:
> 
> :::{image} ../images/span-visualization/2.png
> :screenshot:
> :width: 350px
> :::
> 
> And after clicking on our span, we should see it in detail:
> 
> :::{image} ../images/span-visualization/3.png
> :screenshot:
> :::

Reviewer 2: ```suggestion
You should find your application listed there.

:::{image} ../images/span-visualization/1.png
:screenshot:
:width: 350px
:::

When you open it, go to the **Transactions** tab, where you should see your app's "outermost" spans listed.

:::{image} ../images/span-visualization/2.png
:screenshot:
:width: 350px
:::

And after clicking on the span, you should see it in detail.

:::{image} ../images/span-visualization/3.png
:screenshot:
:::
```

## What’s next? [whats-next]

- This guide uses the minimum configuration options needed to initialize the agent. If you'd like to explore what else you can customize, take a look at the [configuration page](configuration.md).

> - In our [simple example](#hello-world), we've _manually_ sent a span, so we essentially created some [manual instrumentation](manual-instrumentation.md) for our app. That's quite helpful (and flexible), however, the agent can create _automatic instrumentations_ as well! Meaning that, only by initializing the agent, it will start sending telemetry data on your behalf without you having to write code. For more details, refer to [Automatic instrumentation](automatic-instrumentation.md).

Reviewer 3: ```suggestion
- In the [simple example](#hello-world), you've _manually_ sent a span, so you've created some [manual instrumentation](manual-instrumentation.md) for your app. While helpful and flexible, the agent can also create _automatic instrumentations_. This means that by simply initializing the agent, it will start sending telemetry data on your behalf without you having to write code. For more details, refer to [Automatic instrumentation](automatic-instrumentation.md).
```

> - So far, we've used the bare minimum configuration options to initialize the agent, which might be enough for you, however, if you'd like to explore what else you can customize, take a look at the [configuration page](configuration.md).
> 
> - In our [simple example](#hello-world), we've _manually_ sent a span, so we essentially created some [manual instrumentation](manual-instrumentation.md) for our app. That's quite helpful (and flexible), however, the agent can create _automatic instrumentations_ as well! Meaning that, only by initializing the agent, it will start sending telemetry data on your behalf without you having to write code to send anything. Take a look at the [automatic instrumentation](automatic-instrumentation.md) page for more details.
> 
> - [Spans](https://opentelemetry.io/docs/concepts/signals/traces/#spans) are a great way to measure how long some method/part-of-a-method or even some broader transaction that involves multiple methods takes to complete. However, **spans aren't the only type** of [signal](https://opentelemetry.io/docs/concepts/signals/) that you can send using the agent, you can send [logs](https://opentelemetry.io/docs/concepts/signals/logs/) and [metrics](https://opentelemetry.io/docs/concepts/signals/metrics/) too! Take a look at [manual instrumentation](manual-instrumentation.md) for more details.

Reviewer 2: ```suggestion
- This guide uses the minimum configuration options needed to initialize the agent. If you'd like to explore what else you can customize, take a look at the [configuration page](configuration.md).

- In our [simple example](#hello-world), we've _manually_ sent a span, so we essentially created some [manual instrumentation](manual-instrumentation.md) for our app. That's quite helpful (and flexible), however, the agent can create _automatic instrumentations_ as well! Meaning that, only by initializing the agent, it will start sending telemetry data on your behalf without you having to write code. For more details, refer to [Automatic instrumentation](automatic-instrumentation.md).

- [Spans](https://opentelemetry.io/docs/concepts/signals/traces/#spans) are a great way to measure how long some method, _part_ of a method, or even some broader transaction that involves multiple methods takes to complete. However, **spans aren't the only type** of [signal](https://opentelemetry.io/docs/concepts/signals/) that you can send using the agent. You can send [logs](https://opentelemetry.io/docs/concepts/signals/logs/) and [metrics](https://opentelemetry.io/docs/concepts/signals/metrics/) too! For more details, refer to Manual instrumentation](manual-instrumentation.md).
```

Author: I've added a missing '['. Cheers!
